core/src/test/scala/kafka/raft/KafkaMetadataLogTest.scala

```diff
@@ -310,44 +288,109 @@ final class KafkaMetadataLogTest {
 
     log.close()
 
-    val secondLog = buildMetadataLog(tempDir, mockTime, topicPartition)
+    val secondLog = buildMetadataLog(tempDir, mockTime)
 
     assertEquals(snapshotId, secondLog.latestSnapshotId.get)
     assertEquals(snapshotId.offset, secondLog.startOffset)
     assertEquals(snapshotId.epoch, secondLog.lastFetchedEpoch)
     assertEquals(snapshotId.offset, secondLog.endOffset().offset)
     assertEquals(snapshotId.offset, secondLog.highWatermark.offset)
   }
+
+  @Test
+  def testMaxBatchSize(): Unit = {
+    val leaderEpoch = 5
+    val maxBatchSizeInBytes = 16384
+    val recordSize = 64
```

Comment:
Does it matter that maxBatchSizeInBytes is a multiple of the record size? Could you have done 

> buildFullBatch(leaderEpoch, recordSize, maxBatchSizeInBytes + 1) 

to get the same RecordTooLargeException?

Reply:
It's a little tricky to get the alignment perfect after accounting for the overhead of the batch itself.

Reply:
I tried it out just to see if we got lucky on the alignment, but unfortunately we didn't.

Reply:
Thanks for the explanation 👍  I figured it had something to do with the size overhead